coral-spark/src/main/java/com/linkedin/coral/spark/utils/RelDataTypeToHiveTypeStringConverter.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2022 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.spark.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.rel.type.RelRecordType;
+import org.apache.calcite.sql.type.ArraySqlType;
+import org.apache.calcite.sql.type.MapSqlType;
+
+
+/**
+ * Transforms a RelDataType to a Hive type string such that it is parseable and semantically correct.
```

Comment:
Could you add some examples here similar to these in `RelDataTypeToSparkDataTypeStringConverter`?